src/main/java/com/bean00/RequestParser.java

```diff
@@ -3,38 +3,35 @@
 import java.io.BufferedReader;
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
 
 public class RequestParser {
     private BufferedReader in;
+    private HashMap<String, String> requestLineData = new HashMap<>();
 
     public RequestParser(BufferedReader in) {
         this.in = in;
     }
 
     public Request parseRequest() throws IOException {
-        String requestURL = parseRequestLine();
+        parseRequestLine();
```

Comment:
I guess this might be personal preference but I think it's cleaner and more easily understood if you make this a pure function and have a return value. It's harder to understand if they have to go and find out that this function is adding to requestLineData

Reply:
I've split `parseRequestLine` into `getRequestWords`, `getRequestMethod`, and `getRequestURL`, the latter two of which are now pure functions.